### PR TITLE
Add support for .graphqlconfig

### PR DESCRIPTION
**Pull request by RomanGotsiy**
Add support for [`.graphqlconfig`](https://github.com/graphcool/graphql-config)

This PR is backward compatible as it doesn't remove any already existed configuration options. It just adds support for `.graphqlconfig` as a way to configure schema.
That's why I made `schemaJson `, `schemaString` and `schemaJsonFilepath` not required (but they still are mutually exclusive)

cc @jnwng

### CODE REVIEW

**Comment by jnwng** [comment]
awesome @RomanGotsiy! this addresses one of our oldest issues, #8. let me just confirm that this does what we expect with our existing configuration options, and i'll cut a release soon thereafter

**Comment by RomanGotsiy** [comment]
@jnwng amazing! Thanks for the update!

**Comment by schickling** [comment]
Any update on this @jnwng? 🙂

File: src/index.js [hunk 1]
```diff
@@ -120,14 +128,11 @@ export const rules = {
             },
           },
           // schemaJson, schemaJsonFilepath and schemaString are mutually exclusive:
-          oneOf: [{
-            required: ['schemaJson'],
+          allOf: [{
```
Comment (jnwng) [inline src/index.js]: something is a little _off_ about this configuration validation. my assumption is that `projectName` should be mutually exclusive with `schemaJson`, `schemaString`, and `schemaJsonFilepath`, so this configuration validation should reflect that.

additionally, we'd like to make sure that those validations only happen when those configuration properties are present, and if they're not present, it should still be valid.

there's also a section [here](https://github.com/graphcool/eslint-plugin-graphql/blob/0e2581f142915d9a93e9913d3ea5257b7c5b5434/src/index.js#L192-L201) that i think needs to match this configuration as well. can we just extract this configuration key validation and use it for all the rules?

**Comment by RomanGotsiy** [comment]
hey @jnwng 

Just updated PR according to your requirements

**Comment by jnwng** [comment]
thank you @RomanGotsiy!

**Pull request merged**